allennlp/data/vocabulary.py

```diff
@@ -443,7 +444,7 @@ def set_from_file(
             self._token_to_index[namespace] = {}
             self._index_to_token[namespace] = {}
         with codecs.open(filename, "r", "utf-8") as input_file:
-            lines = input_file.read().split("\n")
+            lines = re.split(r"\n|\r\n?", input_file.read())
```

Comment:
What about just

```suggestion
            lines = input_file.read().splitlines()
```

See https://docs.python.org/3/library/stdtypes.html#str.splitlines

Reply:
That was the first idea, but it broke the tests. Because splitlines also split on some table separators, and broke this example:
`vocab_file.write("tricky\x0bchar\n")`

Reply:
Ah, good point.